pkg/connector/users.go

```diff
@@ -200,35 +250,97 @@ func (o *userBuilder) CreateAccount(ctx context.Context, accountInfo *v2.Account
 				}
 			}
 
-			l.Info("user were not found by username, creating database user instead", zap.String("email", email))
+			if user == nil {
+				l.Info("user were not found by username, creating database user instead", zap.String("email", email))
+			}
 		}
 	}
 
 	l.Info("creating database user", zap.String("userId", userId))
-	password, err := crypto.GeneratePassword(ctx, credentialOptions)
-	if err != nil {
-		return nil, nil, nil, uhttp.WrapErrors(codes.Internal, "mongo-db-connector: failed to generate password", err)
+
+	// Get the authentication type from the profile, default to SCRAM-SHA
+	authType, ok := profile["authType"].(string)
+	if authType == "" || !ok {
+		authType = AuthTypeScramSHA
 	}
 
-	defaultDatabase := "admin"
+	// Determine the database name based on authentication type
+	databaseName := getDatabaseNameForAuthType(authType)
+
+	// Build the CloudDatabaseUser with the appropriate auth type fields
+	dbUserRequest := &admin.CloudDatabaseUser{
+		GroupId:      groupId,
+		Username:     username,
+		DatabaseName: databaseName,
+		Roles: &[]admin.DatabaseUserRole{
+			{
+				DatabaseName: databaseNameAdmin,
+				RoleName:     "read",
+			},
+		},
+	}
+
+	var password string
+	var plaintextData []*v2.PlaintextData
+
+	// Set the appropriate authentication type fields
+	// See: https://www.mongodb.com/docs/api/doc/atlas-admin-api-v2/operation/operation-createdatabaseuser
+	switch authType {
+	case AuthTypeScramSHA:
+		// SCRAM-SHA requires a password
+		password, err = crypto.GeneratePassword(ctx, credentialOptions)
+		if err != nil {
+			return nil, nil, nil, uhttp.WrapErrors(codes.Internal, "mongo-db-connector: failed to generate password", err)
+		}
+		dbUserRequest.Password = &password
+		plaintextData = []*v2.PlaintextData{
+			{
+				Name:        "password",
+				Description: "The password for the database user",
+				Schema:      "text/plain",
+				Bytes:       []byte(password),
+			},
+		}
+
+	case AuthTypeAWSIAMUser:
```

Comment:
To make these shorter, you could have a little helper function like

```golang
func strPtr(s string) *string {
	return &s
}
```

Then you could do these on one line:
```golang
switch authType {
	case AuthTypeAWSIAMUser:
		// AWS IAM User authentication - username must be an AWS ARN
		dbUserRequest.AwsIAMType = strPtr(dbTypeUser)
	case AuthTypeX509Customer:
		// Customer-managed X.509 certificate - username must be RFC 2253 Distinguished Name
		dbUserRequest.X509Type = strPtr("CUSTOMER")
	case AuthTypeX509Managed:
		// MongoDB Atlas-managed X.509 certificate - username must be RFC 2253 Distinguished Name
		dbUserRequest.X509Type = strPtr("MANAGED")
```